ops/model.py

```diff
@@ -3289,7 +3289,7 @@ def secret_get(self, *,
         args: List[str] = []
         if id is not None:
             args.append(id)
-        if label is not None:
+        elif label is not None:
```

Comment:
I'm concerned that this isn't correct, at least for secret observers: when calling `get_secret`, the observer looks it up by ID, but the label given is *applied* as a label when `get_secret(id=..., label=label_to_apply)` is called. See the docs for `Model.get_secret`:

```
    def get_secret(self, *, id: Optional[str] = None, label: Optional[str] = None) -> 'Secret':
        """Get the :class:`Secret` with the given ID or label.

        The caller must provide at least one of `id` (the secret's locator ID)
        or `label` (the charm-local "name").

        If both are provided, the secret will be fetched by ID, and the
        secret's label will be updated to the label provided. Normally secret
        owners set a label using ``add_secret``, whereas secret observers set
        a label using ``get_secret`` (see an example at :attr:`Secret.label`).
```

Let's discuss with @wallyworld to see how we should solve the underlying issue.